storage/innobase/row/row0ins.cc

```diff
@@ -778,7 +780,7 @@ row_ins_foreign_report_err(
 		dtuple_print(ef, entry);
 	}
 	fputs("\nBut in child table ", ef);
-	ut_print_name(ef, trx, foreign->foreign_table_name);
+	fputs(fk_table_str.c_str(), ef);
 	fprintf(ef, ", in index %s", foreign->foreign_index->name());
```

Comment:
Please, let’s avoid using any `std::string` and use a single `my_printf_error` or similar for this. See how the function `dict_table_open_failed()` is displaying table names.

Reply:
Yes, it was already suboptimal, and since the time the code was originally written, we are aware of a more efficient way of displaying InnoDB table names. One thing that I overlooked yesterday is the conversion from `my_charset_filename` to `system_charset_info` (UTF-8). A test case for this must involve table names that include some non-alphanumeric characters, such as `-` or `$`. The test case also needs to search for these messages in the `SHOW ENGINE INNODB STATUS` output.